server/src/com/treode/server/Resource.scala

```diff
@@ -97,7 +97,30 @@ class Resource (host: HostId, store: SchematicStore) extends Service [Request, R
         respond (req, Status.PreconditionFailed)
     }}
 
+  def batch_write (req: Request): Async [Response] = {
+    val tx = req.transactionId (host)
+    val ct = req.conditionTxClock (TxClock.now)
+    try {
+      store.batch_write(tx, ct, req)
+      .map [Response] { vt =>
+        val rsp = req.response
+        rsp.status = Status.Ok
+        rsp.headerMap.add ("Value-TxClock", vt.toString)
+        rsp
+      }
+      .recover {
+        case _: StaleException => {
+          respond (req, Status.PreconditionFailed)
+        }
+      }
+    } catch {
+      case v: Throwable => {
```

Comment:
Good catch. You should throw an error that is as informative as possible. So that does mean checking manually. That way you can say, "op" is required, or "value is required for create/update".